synapse/models/telco.py

```diff
@@ -113,62 +148,183 @@ def imeicsum(text):
 
     return str(chek)
 
-class ImeiType(DataType):
-    '''
-    https://en.wikipedia.org/wiki/International_Mobile_Equipment_Identity
-    '''
+class Imsi(s_types.Type):
```

Comment:
This one also gives me pause about subclassing s_types.Int, but it seems like it does just enough differently that it shouldn't. :)